docs/book/src/tasks/experimental-features/cluster-class/change-clusterclass.md

```diff
@@ -1,32 +1,190 @@
 | workers.machineDeployments[].bootstrap.ref      | If the referenced template has changes only in metadata labels or annotations, the corresponding BootstrapTemplates are updated (in place update).<br /> <br />If the referenced template has changes in the spec:<br />  -  Corresponding BootstrapTemplate are rotated (create new, delete old). <br />  - Corresponding MachineDeployments objects are updated with the reference to the newly created template (in place update). <br />  - The corresponding worker machines are updated accordingly (rollout)                        |
 | workers.machineDeployments[].infrastructure.ref | If the referenced template has changes only in metadata labels or annotations, the corresponding InfrastructureMachineTemplates are updated (in place update). <br /> <br />If the referenced template has changes in the spec:<br />  -  Corresponding InfrastructureMachineTemplate are rotated (create new, delete old).<br />  -  Corresponding MachineDeployments objects are updated with the reference to the newly created template (in place update). <br />  - The corresponding worker Machines are updated accordingly (rollout) |
 
+### How the topology controller reconciles template fields
+
+Advanced users might be interested in understanding how the topology reconciler
+enforces the values defined in the ClusterClass templates into the topology owned 
+objects in a Cluster.
+
+A simple way to understand this is to `kubectl get -o json` templates referenced in a ClusterClass;
+then you can consider the topology reconciler to be authoritative on all the values
+under `spec`. Being authoritative means that the user cannot change manually those values in
+the object derived from the template in a specific Cluster (and if he does so the value gets reconciled
+to the value defined in the ClusterClass).
+
+<aside class="note">
+<h1>What about patches</h1>
+
+The considerations above apply also when using patches, with the only difference that the
+authoritative fields should be determined by applying patches on top of the `kubectl get -o json` output. 
+
+</aside>
 
-Note: In case a provider supports in place template mutations, the Cluster API topology controller will adapt to them at the next reconciliation, but the system is not watching for those specific changes. When the underlying template is updated in this way the changes may not be reflected immediately, but will be put in place at the next full reconciliation. The maximum time for the next reconciliation to take place is related to the CAPI controller sync period - 10 minutes by default.
+A corollary of the behaviour described above is that it is technically possible to change non-authoritative
+fields in the object derived from the template in a specific Cluster, but we advise to use the possibility 
+to do ad-hoc changes in generated objects only as workaround, while instead it should always be
```

Comment:
```suggestion
fields in the object derived from the template in a specific Cluster, but we advise against using the possibility 
or making ad-hoc changes in generated objects unless otherwise needed for a workaround. It is always
```
nit: To enforce that we don't recommend doing direct changes. Feel free to ignore if you prefer the current wording.